nixos/modules/system/boot/systemd.nix

```diff
@@ -411,30 +411,6 @@ in
   ###### implementation
 
   config = {
-
-    warnings = concatLists (
-      mapAttrsToList
-        (name: service:
-          let
-            type = service.serviceConfig.Type or "";
-            restart = service.serviceConfig.Restart or "no";
-            hasDeprecated = builtins.hasAttr "StartLimitInterval" service.serviceConfig;
-          in
-            concatLists [
-              (optional (type == "oneshot" && (restart == "always" || restart == "on-success"))
-                "Service '${name}.service' with 'Type=oneshot' cannot have 'Restart=always' or 'Restart=on-success'"
-              )
-              (optional hasDeprecated
-                "Service '${name}.service' uses the attribute 'StartLimitInterval' in the Service section, which is deprecated. See https://github.com/NixOS/nixpkgs/issues/45786."
-              )
-              (optional (service.reloadIfChanged && service.reloadTriggers != [])
-                "Service '${name}.service' has both 'reloadIfChanged' and 'reloadTriggers' set. This is probably not what you want, because 'reloadTriggers' behave the same whay as 'restartTriggers' if 'reloadIfChanged' is set."
```

Comment:
I do not know what happens when both is set. Never the less, I do not understand this warning, since I do not know how restartTriggers behave when reloadIfChanged is set.